addons/website_event_meet/controllers/main.py

```diff
@@ -0,0 +1,163 @@
+            "default_lang_code": request.env.user.lang,
+            "open_room_id": int(open_room_id) if open_room_id else None,
+            # environment
+            "is_event_manager": request.env.user.has_group("event.group_event_manager"),
+        }
+
+    @http.route(["/event/create_meeting_room"], type="http", auth="public", methods=["POST"], website=True)
+    def create_meeting_room(self, **post):
+        name = post.get("name")
+        summary = post.get("summary")
+        target_audience = post.get("audience")
+        lang_code = post.get("lang_code")
+        max_capacity = post.get("capacity")
+        event_id = int(post.get("event"))
+
+        # get the record to be sure they really exist
+        event = request.env["event.event"].browse(event_id).exists()
+        lang = request.env["res.lang"].search([("code", "=", lang_code)], limit=1)
+
+        if not event or not event.can_access_from_current_website():
+            raise Forbidden()
+
+        if not event.website_published or not lang or max_capacity == "no_limit":
+            raise Forbidden()
+
+        _logger.info("New meeting room (%s) create by %s" % (name, request.httprequest.remote_addr))
+
+        meeting_room = request.env["event.meeting.room"].sudo().create(
+            {
+                "name": name,
+                "summary": summary,
+                "target_audience": target_audience,
+                "is_pinned": False,
+                "event_id": event.id,
+                "room_lang_id": lang.id,
+                "room_max_capacity": max_capacity,
+            },
+        )
+
+        return redirect("/event/%s/meeting_rooms?open_room_id=%i" % (slug(event), meeting_room.id))
+
+    @http.route(["/event/active_langs"], type="json", auth="public")
+    def active_langs(self):
+        return request.env["res.lang"].sudo().get_installed()
+
+    # ------------------------------------------------------------
+    # ROOM PAGE VIEW
+    # ------------------------------------------------------------
+
+    @http.route(["/event/<model('event.event'):event>/meeting_room/<model('event.meeting.room'):meeting_room>"], type="http",
+                auth="public", website=True, sitemap=True)
+    def event_meeting_room(self, event, meeting_room, **post):
+        """Display the meeting room frontend view.
+
+        :param event: Event for which we display the meeting rooms
+        :param meeting_room: Meeting Room to display
+        """
+        if not event.can_access_from_current_website() or meeting_room not in event.sudo().meeting_room_ids:
+            raise NotFound()
+
+        try:
+            meeting_room.check_access_rule('read')
+        except exceptions.AccessError:
+            raise Forbidden()
+
+        meeting_room = meeting_room.sudo()
+
+        return request.render(
+            "website_event_meet.event_meet_main",
+            self._event_meet_get_values(event, meeting_room)
```

Comment:
meeting_room.sudo() (or maybe it's because it will be more hidden than sudoing meeting_room in a separated line?)